src/Chainweb/Pact/PactService/Checkpointer.hs

```diff
@@ -530,7 +530,7 @@ lookupBlockHeader bhash ctx = do
       then return cur
       else do
         bhdb <- asks _psBlockHeaderDb
-        liftIO $! lookupM bhdb bhash `catchAllSynchronous` \e ->
+        liftIO $! lookupM bhdb bhash `catchAny` \e ->
```

Comment:
Are we sure that `BlockHeader` deserialization is forced nf in the result of `lookupM`. Realistically, we only expect failures from `IO` actions, like broken db connections or missing keys. But I wonder if corrupted db data could hide in an unevaluated thunk in the result.

Reply:
Good point, I have added some extra strictness to `decodeBlockHashRecord` for this but yes there are no other errors that are not thrown during deserialization.